system_metrics_collector/src/topic_statistics_collector/received_message_period.hpp

```diff
@@ -23,9 +23,13 @@
 #include "system_metrics_collector/collector.hpp"
 
 #include "rcl/time.h"
-#include "rclcpp/clock.hpp"
 
 
+namespace
+{
+constexpr const int64_t kUninitializedTime{0};
```

Comment:
need to be defined in `received_message_period.cpp`

Reply:
IMO: this should instead be moved into `namespace topic_statistics_collector`

Because `kUninitializedTime` is a numeric type, duplication in the code shouldn't matter, especially since this is a private header. I'm not fully confident with this second opinion so I'm going to double check it.

Reply:
I looked up the first comment, we should not use [anonymous namepaces](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-unnamed) in headers.

Reply:
yes, that too.